src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  setOpenedTodo: (todo: Todo | null) => void;
+  setShowModal: (showModal: boolean) => void;
+  openedTodo: Todo | null;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  setOpenedTodo,
+  setShowModal,
+  openedTodo,
+}) => {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+
+  useEffect(() => {
+    if (!openedTodo) {
+      return;
+    }
+
+    getUser(openedTodo.userId).then(userInfo => {
+      setUsername(userInfo.name);
+      setEmail(userInfo.email);
+    });
+  }, []);
```

Comment:
The dependency array for useEffect is empty, which means this effect will only run once when the component mounts. If 'openedTodo' changes, the effect will not re-run. Consider adding 'openedTodo' to the dependency array to ensure the effect runs whenever 'openedTodo' changes.